bgpd/bgp_nexthop.c

```diff
@@ -528,35 +532,33 @@ static void bgp_show_nexthops(struct vty *vty, struct bgp *bgp, int detail)
 
 		for (rn = bgp_table_top(bgp->nexthop_cache_table[afi]); rn;
 		     rn = bgp_route_next(rn)) {
-			if ((bnc = rn->info) != NULL) {
-				if (CHECK_FLAG(bnc->flags, BGP_NEXTHOP_VALID)) {
-					vty_out(vty,
-						" %s valid [IGP metric %d], #paths %d\n",
-						inet_ntop(rn->p.family,
-							  &rn->p.u.prefix, buf,
-							  sizeof(buf)),
-						bnc->metric, bnc->path_count);
-
-					if (!detail)
-						continue;
-
-					bgp_show_nexthops_detail(vty, bgp, bnc);
-
-				} else {
-					vty_out(vty, " %s invalid\n",
-						inet_ntop(rn->p.family,
-							  &rn->p.u.prefix, buf,
-							  sizeof(buf)));
-					if (CHECK_FLAG(bnc->flags,
-						       BGP_NEXTHOP_CONNECTED))
-						vty_out(vty,
-							"  Must be Connected\n");
-				}
-				tbuf = time(NULL)
-				       - (bgp_clock() - bnc->last_update);
-				vty_out(vty, "  Last update: %s", ctime(&tbuf));
-				vty_out(vty, "\n");
+			bnc = bgp_nexthop_from_node(rn);
+			if (!bnc)
+				continue;
```

Comment:
I have a very slight preference to maintaining the previous if structure, just so that it makes it clear what actually is changing with the PR.  This said, I don't feel strongly about this -- so I can live with the current code if you don't want to change it back.

Reply:
I'll see what I can do if you don't mind me tacking on an additional commit in here